04.02 - working with data/assigment1/src/js/main.js

```diff
@@ -1,42 +1,175 @@
 import { contacts } from '../data/contacts.js';
 
-console.log(contacts);
+//console.log(contacts.results);
 
 
 // practice of shallow and deep copying of an object
 
-let person = {
-    name: 'Alex',
-    email: 'alex@alex.com',
-    address: {
-        city: 'Tel-Aviv',
-        street: 'Herzel',
-        house: 15,
-        flat: 1
-    },
-    logPersonName: function(){
-        console.log(this.name);
+// let person = {
+//     name: 'Alex',
+//     email: 'alex@alex.com',
+//     address: {
+//         city: 'Tel-Aviv',
+//         street: 'Herzel',
+//         house: 15,
+//         flat: 1
+//     },
+//     logPersonName: function(){
+//         console.log(this.name);
+//     }
+// }
+
+// let person1 = {...person};
+
+// person1.name = 'Alexey';
+// person1.address.city = 'Haifa';
+// person1.logPersonName = function(){
+//     console.log('Hey, ' + this.name);
+// }
+
+// console.log(person);
+
+// let person2 = JSON.parse(JSON.stringify(person));
+// console.log(person2);
+
+// let person3 = Object.assign({}, person);
+// console.log(person3);
+
+// person3.name = 'Alexander';
+// person3.address.city = 'Eilat';
+// person3.logPersonName = function(){
+//     console.log('Hey-hey, ' + this.name);
+// }
+
+
+// # Array Order Methods 
+
+// === task1 ===
+// 1. Write a function that takes the array of contacts and returns a new array sorted by last name. 
+
+const sortContactsByLastName = function(contactsArr){
+    const contactsArrCopy = JSON.parse(JSON.stringify(contactsArr));
+    contactsArrCopy.sort((a,b)=> a.name.last > b.name.last ? 1 : -1);
+    return contactsArrCopy;
+}
+
+// sortContactsByLastName(contacts.results).forEach((el)=> console.log(el.name.last));
+
+
+
+// === task 2 ===
+// 2. Write a function to return the array of contacts in reverse order. Do not use the built-in reverse() method. 
+
+const getReversedArr = function(arr){
+    const reversedArr = [];
+    const cnt = arr.length;
+    for (let index = 0; index < arr.length; index++) {
+        reversedArr.push(arr[cnt - index - 1]);
+    };
+    return reversedArr;
+}
+
+// getReversedArr(contacts.results).forEach((item) => console.log(item.email));
+
+
+
+// task 3
+// 3. Write a function that returns the first 5 contacts from the sorted list (by last name).
+
+const returnTopNContactsByLastName = function(contactsArr, Num){
```

Comment:
nice dynamic definiton!
**challenge**
do we need validation for num?
if so what kind?